src/mlpack/core/optimizers/sgd/update_policies/momentum_update.hpp

```diff
@@ -0,0 +1,110 @@
+/**
+ * @file momentum_update.hpp
+ * @author Arun Reddy
+ *
+ * Momentum update for Stochastic Gradient Descent.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_OPTIMIZERS_SGD_MOMENTUM_UPDATE_HPP
+#define MLPACK_CORE_OPTIMIZERS_SGD_MOMENTUM_UPDATE_HPP
+
+namespace mlpack {
+namespace optimization {
+
+/**
+ * Momentum update policy for Stochastic Gradient Descent (SGD).
+ *
+ * Learning with SGD can sometimes be slow.  Using momentum update for parameter
+ * learning can accelerate the rate of convergence, specifically in the cases
+ * where the surface curves much more steeply(a steep hilly terrain with high
+ * curvature).  The momentum algorithm introduces a new velocity vector \f$ v
+ * \f$ with the same dimension as the parameter \f$ A \f$.  Also it introduces a
+ * new decay hyperparameter momentum \f$ mu \in (0,1] \f$.  The following update
+ * scheme is used to update SGD in every iteration:
+ *
+ * \f[
+ * v = mu*v - \alpha \nabla f_i(A)
+ * A_{j + 1} = A_j + v
+ * \f]
+ *
+ * where \f$ \alpha \f$ is a parameter which specifies the step size.  \f$ i \f$
+ * is chosen according to \f$ j \f$ (the iteration number).  Common values of
+ * \f$ mu \f$ include 0.5, 0.9 and 0.99.  Typically it begins with a small value
+ * and later raised.
+ *
+ * For more information, please refer to the Section 8.3.2 of the following book
+ *
+ * @code
+ * @book{Goodfellow-et-al-2016,
+ *  title={Deep Learning},
+ *  author={Ian Goodfellow and Yoshua Bengio and Aaron Courville},
+ *  publisher={MIT Press},
+ *  note={\url{http://www.deeplearningbook.org}},
+ *  year={2016}
```

Comment:
I'm a fan of giving credit to the first paper, that proposed the method, which I guess is "Learning representations by back-propagating errors" by Rumelhart et al. if you like we can cite both papers/books. What do you think?